_docs/pages/instrumentation/common-use-cases.md

```diff
@@ -253,3 +253,69 @@ if request.get('debug'):
         tags={tags.SAMPLING_PRIORITY: 1}
     )
 ```
+
+### Tracing Messaging Scenarios
+
+There are two messaging styles that should be handled, Message Queues and Publish/Subscribe (Topics).
+
+The main distinction between these styles is that a message written to a queue can only be consumed by **at most** one consumer, whereas a message published on a topic can be consumed by zero or more subscribers. The other point to note is that although a message can be sent/published by a producer, that does not guarantee that it will be consumed - messaging systems are asynchronous and therefore the producer has no visibility of when or if the message was delivered.
+
+From a tracing perspective, the messaging style is not important, only that the span context associated with the producer is propagated to the zero or more consumers of the message. It is then the responsibility of the consumer(s) to create a span to encapsulate processing of the consumed message and establish a _FollowsFrom_ reference to the propagated span context.
```

Comment:
should there be a difference between FollowsFrom used to fire&forget a cache write in the same process vs. the messaging case? I think depending on the situation some users may want the messaging consumer to start a new trace and link it to the parent trace (e.g. if the parent was a part of an http request), while in other cases (pure messaging based system) they may want the same trace. Are we suggesting that the tracing system must make that decision on its own, or should we have another reference type or some other indicator from the instrumentation whether a single trace or linked traces are desired?  cc @bhs